src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultModelBindingContext.cs

```diff
@@ -224,13 +227,15 @@ public override ModelBindingResult Result
             var propertyFilterProvider = bindingInfo?.PropertyFilterProvider ?? metadata.PropertyFilterProvider;
 
             var bindingSource = bindingInfo?.BindingSource ?? metadata.BindingSource;
+            var loggerFactory = actionContext.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
 
             return new DefaultModelBindingContext()
             {
                 ActionContext = actionContext,
                 BinderModelName = binderModelName,
                 BindingSource = bindingSource,
                 PropertyFilter = propertyFilterProvider?.PropertyFilter,
+                _loggerFactory = loggerFactory,
```

Comment:
Setting a `private` filed is just ugly; add this to the constructor. Better, if the `DefaultModelBindingContext` must have an `ILoggerFactory`, make it `public` in `ModelBindingContext` and settable in `DefaultModelBindingContext`.

But, are the few messages `CompositeValueProvider` logs worth the complications here, in `FilterValueProvider`, and everywhere `CompositeValueProvider` instances are created? If we need this, similar-value messages can be logged in the model binders.